core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -108,7 +119,7 @@ protected String name() {
 
     try {
       if (!fs.exists(nsPath) || !fs.isDirectory(nsPath)) {
-        throw new NotFoundException("Unknown namespace " + namespace);
+        throw new NoSuchNamespaceException("Namespace does not exist: " + namespace);
```

Comment:
This exception class supports format strings and args, so this should use `%s` and pass namespace as an argument.